### PR TITLE
[Eve] Improve Eve Energy support

### PR DESCRIPTION
**Pull request by Timac**
These changes improve the Eve Energy support

- A subdriver has been implemented
- It implements the switch, powerMeter, energyMeter and powerConsumptionReport capabilities
- It supports the refresh capability
- It reads with a timer each minute the current Watt and Watt Accumulated values from custom attributes
- The powerMeter and energyMeter values are updated each values or on a refresh
- A powerConsumptionReport is sent each 10 minutes by comparing the previous Watt Accumulated value with the current Watt Accumulated value

### CODE REVIEW

File: drivers/SmartThings/matter-switch/profiles/eve-energy.yml
Comment (greens) [inline drivers/SmartThings/matter-switch/profiles/eve-energy.yml:1]: filename is incorrect, you'll want to `git mv` it

File: drivers/SmartThings/matter-switch/src/eve-energy/init.lua
Comment (greens) [inline drivers/SmartThings/matter-switch/src/eve-energy/init.lua:1]: missing our copyright notice

File: drivers/SmartThings/matter-switch/src/eve-energy/init.lua [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+end
+
+
+-------------------------------------------------------------------------------------
+-- Eve Energy Handler
+-------------------------------------------------------------------------------------
+
+local function matter_handler(driver, device, response_block)
+  log.info(string.format("Fallback handler for %s", response_block))
+end
```
Comment (greens) [inline drivers/SmartThings/matter-switch/src/eve-energy/init.lua:177]: this behavior will be inherited from the base driver, so there's no reason to include it here

**Review by greens — COMMENTED** [review]
We'd like to see some tests written for these changes.

File: drivers/SmartThings/matter-switch/profiles/eve-energy.yml
Comment (Timac) [reply drivers/SmartThings/matter-switch/profiles/eve-energy.yml:1]: Done

**Comment by Timac** [comment]
> We'd like to see some tests written for these changes.

I added the file `test_eve_energy.lua` with really basic tests. At the moment, I don’t know how to properly test the custom attributes on the custom cluster. Maybe @greens  you might have some ideas?

The function `requestData` is called each minute with the timer to read the custom attribute on the custom cluster: 

```
local PRIVATE_CLUSTER_ID = 0x130AFC01

local PRIVATE_ATTR_ID_WATT = 0x130A000A
local PRIVATE_ATTR_ID_WATT_ACCUMULATED = 0x130A000B

[…]

local function requestData(device)
  -- Update the on/off status
  device:send(clusters.OnOff.attributes.OnOff:read(device))

  -- Update the Watt usage
  device:send(cluster_base.read(device, 0x01, PRIVATE_CLUSTER_ID, PRIVATE_ATTR_ID_WATT, nil))

  -- Update the energy consumption
  device:send(cluster_base.read(device, 0x01, PRIVATE_CLUSTER_ID, PRIVATE_ATTR_ID_WATT_ACCUMULATED, nil))
end
```

**Comment by greens** [comment]
@Timac I believe you'd want to use 

```
test.timer.__create_and_queue_test_time_advance_timer(60, "interval")
```

Similarly to how it is used in this zigbee test: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-contact/src/test/test_zigbee_contact.lua#L141

And then, you'd want to queue up `expect_send`s passing `read`s the exact way you've constructed them in the driver.

**Comment by Timac** [comment]
> @Timac I believe you'd want to use
> 
> ```
> test.timer.__create_and_queue_test_time_advance_timer(60, "interval")
> ```
> 
> Similarly to how it is used in this zigbee test: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-contact/src/test/test_zigbee_contact.lua#L141
> 
> And then, you'd want to queue up `expect_send`s passing `read`s the exact way you've constructed them in the driver.

That worked. Thanks a lot!

File: drivers/SmartThings/matter-switch/src/test/test_eve_energy.lua [hunk 2]
```diff
@@ -0,0 +1,278 @@
+-- Copyright 2022 SmartThings
```
Comment (greens) [inline drivers/SmartThings/matter-switch/src/test/test_eve_energy.lua:1]: nit: year

**Review by greens — APPROVED** [review]
looks good

File: drivers/SmartThings/matter-switch/src/test/test_eve_energy.lua (hunk 2 above)
Comment (Timac) [reply drivers/SmartThings/matter-switch/src/test/test_eve_energy.lua]: Nice catch. Fixed!

**Pull request merged**